extensions/spark/kyuubi-spark-authz/src/main/scala/org/apache/kyuubi/plugin/spark/authz/ranger/RuleApplyDataMasking.scala

```diff
@@ -0,0 +1,107 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one or more
+ * contributor license agreements.  See the NOTICE file distributed with
+ * this work for additional information regarding copyright ownership.
+ * The ASF licenses this file to You under the Apache License, Version 2.0
+ * (the "License"); you may not use this file except in compliance with
+ * the License.  You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.plugin.spark.authz.ranger
+
+import org.apache.hadoop.security.UserGroupInformation
+import org.apache.spark.sql.SparkSession
+import org.apache.spark.sql.catalyst.expressions.{Alias, NamedExpression, SubqueryExpression}
+import org.apache.spark.sql.catalyst.plans.logical.{Command, LogicalPlan}
+import org.apache.spark.sql.catalyst.rules.Rule
+
+import org.apache.kyuubi.plugin.spark.authz.ObjectType._
+import org.apache.kyuubi.plugin.spark.authz.OperationType.OperationType
+import org.apache.kyuubi.plugin.spark.authz.ranger.AccessType._
+import org.apache.kyuubi.plugin.spark.authz.serde._
+import org.apache.kyuubi.plugin.spark.authz.util.AuthZUtils._
+import org.apache.kyuubi.plugin.spark.authz.util.DataMaskingMarker
+
+class RuleApplyDataMasking(spark: SparkSession) extends Rule[LogicalPlan] {
+  private val parse = spark.sessionState.sqlParser.parseExpression _
+
+  override def apply(plan: LogicalPlan): LogicalPlan = {
+    val ugi = getAuthzUgi(spark.sparkContext)
+    val opType = operationType(plan)
+
+    plan match {
+      case marker: DataMaskingMarker => marker
+      case cmd if isKnownTableCommand(cmd) =>
+        val tableCommandSpec = getTableCommandSpec(cmd)
+        val queries = tableCommandSpec.queries(cmd)
+        cmd.mapChildren {
+          case marker: DataMaskingMarker => marker
+          case query if queries.contains(query) && query.resolved =>
+            applyDataMasking(query, ugi, opType)
+          case o => o
+        }
+      case cmd: Command if cmd.childrenResolved =>
+        cmd.mapChildren(applyDataMasking(_, ugi, opType))
+      case cmd: Command => cmd
```

Comment:
Is there ut for commands with masking rules applied in V2?

Reply:
yes